server/lib/scene/scene.actions.js

```diff
@@ -110,6 +110,47 @@ const actionsFunc = {
       }
     });
   },
+  [ACTIONS.LIGHT.BLINK]: async (self, action, scope) => {
+    const { blinkingSpeed, blinkingTime } = action;
+    let blinkingInterval;
+    switch (blinkingSpeed) {
+      case 'slow':
+        blinkingInterval = 1000;
+        break;
+      case 'medium':
+        blinkingInterval = 500;
+        break;
+      case 'fast':
+        blinkingInterval = 200;
+        break;
+      default:
+        blinkingInterval = 200;
+        break;
+    }
+    await Promise.map(action.devices, async (deviceSelector) => {
+      try {
+        const device = self.stateManager.get('device', deviceSelector);
+        const deviceFeature = getDeviceFeature(
+          device,
+          DEVICE_FEATURE_CATEGORIES.LIGHT,
+          DEVICE_FEATURE_TYPES.LIGHT.BINARY,
+        );
+        const oldValue = deviceFeature.last_value;
+        let newValue = 0;
+        let timerId = setTimeout(function blink() {
+          newValue = 1 - newValue;
+          self.device.setValue(device, deviceFeature, newValue);
+          timerId = setTimeout(blink, blinkingInterval);
+        }, blinkingInterval);
+        setTimeout(() => {
```

Comment:
I think we need to make sure that this action is blocking 

Because right now the logic in scene actions is that each action is blocking and we are going to the next block of action only when all actions in the preceding block are finished

Reply:
Fixed